File: TelerikUIExamples/TelerikUIExamples/DataFormGettingStarted.m
//
//  DataFormGettingStarted.m
//  TelerikUIExamples
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

#import "DataFormGettingStarted.h"
#import "PersonalInfo.h"

@implementation DataFormGettingStarted
{
    TKDataFormEntityDataSource *_dataSource;
    PersonalInfo *_personalInfo;
}

// >> dataform-ctrl-setup
- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _personalInfo = [PersonalInfo new];
    self.dataForm.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    self.dataForm.backgroundColor = [UIColor colorWithRed:0.937 green:0.937 blue:0.960 alpha:1.0];

    _dataSource = [[TKDataFormEntityDataSource alloc] initWithObject:_personalInfo];
    _dataSource[@"password"].hintText = @"Ask every time";
    _dataSource[@"password"].editorClass = [TKDataFormPasswordEditor class];
    
    _dataSource[@"protocol"].valuesProvider = @[@"L2TP", @"PPTP", @"IPSec"];
    _dataSource[@"encryptionLevel"].valuesProvider = @[@"FIPS Compliant", @"High", @"Client Compatible", @"Low"];
    
    [_dataSource addGroupWithName:@"  " propertyNames:@[ @"protocol" ]];
    [_dataSource addGroupWithName:@" " propertyNames:@[ @"details", @"server", @"account", @"secure", @"password", @"encryptionLevel", @"sendAllTraffic" ]];

    self.dataForm.dataSource = _dataSource;
    self.dataForm.commitMode = TKDataFormCommitModeOnLostFocus;
    self.dataForm.groupSpacing = 20;
}
// << dataform-ctrl-setup

#pragma mark TKDataFormDelegate

// >> dataform-delegate
- (void)dataForm:(TKDataForm *)dataForm updateEditor:(TKDataFormEditor *)editor forProperty:(TKEntityProperty *)property
{
    TKGridLayoutCellDefinition *feedbackDef = [editor.gridLayout definitionForView:editor.feedbackLabel];
    [editor.gridLayout setHeight:0 forRow:[feedbackDef.row integerValue]];
    
    if ([property.name isEqualToString:@"protocol"]) {
        editor.style.textLabelDisplayMode = TKDataFormEditorTextLabelDisplayModeHidden;
        TKGridLayoutCellDefinition *textLabelDef = [editor.gridLayout definitionForView:editor.textLabel];
        [editor.gridLayout setWidth:0 forColumn:[textLabelDef.column integerValue]];
    }
}
// << dataform-delegate

- (void)dataForm:(TKDataForm *)dataForm updateGroupView:(TKEntityPropertyGroupView *)groupView forGroupAtIndex:(NSUInteger)groupIndex {
    if (groupIndex == 1) {
        groupView.editorsContainer.layout = [[TKGridLayout alloc] init];
    }
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/DataFormGettingStarted.swift
//
//  DataFormGettingStarted.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import UIKit

class DataFormGettingStarted: TKDataFormViewController {

    let dataSource = TKDataFormEntityDataSource()
    let personalInfo = PersonalInfo()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.dataForm.backgroundColor = UIColor(red: 0.937, green: 0.937, blue: 0.960, alpha: 1.0)
        dataSource.sourceObject = personalInfo

        dataSource["password"].hintText = "Ask every time"
        dataSource["password"].editorClass = TKDataFormPasswordEditor.self
        dataSource["infoProtocol"].valuesProvider = ["L2TP", "PPTP", "IPSec"]
        dataSource["encryptionLevel"].valuesProvider = [ "FIPS Compliant", "High", "Client Compatable", "Low" ]
        
        dataSource.addGroupWithName(" ", propertyNames: [ "infoProtocol" ])
        dataSource.addGroupWithName(" ", propertyNames: [ "details", "server", "account", "secure", "password", "encryptionLevel", "sendAllTraffic" ])
        
        self.dataForm.dataSource = dataSource
        self.dataForm.commitMode = TKDataFormCommitMode.OnLostFocus
        self.dataForm.groupSpacing = 20
    }
    
    //MARK: - TKDataFormDelegate
    
    override func dataForm(dataForm: TKDataForm, updateEditor editor: TKDataFormEditor, forProperty property: TKEntityProperty) {
        let feedbackDef = editor.gridLayout.definitionForView(editor.feedbackLabel)
        editor.gridLayout.setHeight(0, forRow: feedbackDef.row.integerValue)
        
        if property.name == "infoProtocol" {
            editor.style.textLabelDisplayMode = TKDataFormEditorTextLabelDisplayMode.Hidden
            let textLabelDef = editor.gridLayout.definitionForView(editor.textLabel)
            editor.gridLayout.setWidth(0, forColumn: textLabelDef.column.integerValue)
        }
        
        if editor.isKindOfClass(TKDataFormTextFieldEditor.self) && property.name != "password" {
            property.hintText = "Required"
        }
    }
    
    override func dataForm(dataForm: TKDataForm, updateGroupView groupView: TKEntityPropertyGroupView, forGroupAtIndex groupIndex: UInt) {
        groupView.titleView.hidden = true
    }
}
